### ISSUE TITLE
Error: could not find template for MNI152NLin6Sym in bold preprocessing, but anat works fine

### ISSUE THREAD

**Issue by antotorov** [poldracklab/fmriprep#1809]
Hello!

I ran into an error running fmriprep-docker with the options

> --output-spaces 'MNI152NLin6Asym:res-2' 'T1w' 'MNI152NLin6Sym:res-2'

The error message I got contains

> RuntimeError: Could not find template "MNI152NLin6Sym" with specs={'suffix': 'T1w', 'desc': None, 'atlas': None, 'resolution': '2'}. Please revise your template argument.

I first thought some template was missing but the anat part of the preprocessing seems to have worked fine (in particular I have all the outputs I was expecting like `sub-001_space-MNI152NLin6Sym_desc-preproc_T1w.nii.gz` etc).  For the bold part I have the expected `*_space-T1w_desc-preproc_bold.nii.gz` and `*_space-MNI152NLin6Asym_desc-preproc_bold.nii.gz` files, but no `*_space-MNI152NLin6Sym_desc-preproc_bold.nii.gz`.

Many thanks in advance for your help!

**Comment by effigies** [comment]
Hi, can you please provide your full command and what environment you're running in?

**Comment by mgxd** [comment]
it looks like there is no `res-2` for the `MNI152NLin6Sym` output-space
https://github.com/templateflow/tpl-MNI152NLin6Sym/tree/9678ab9b6349f2d25c3248472176d7eb94321bfd

**Comment by antotorov** [comment]
@effigies : sure, I am on MACOS, runing fmriprep 1.5.0 through the docker wrapper fmriprep-docker. The full command is

`fmriprep-docker "$BIDSDATA "$DERIVATIVES "participant" --fs-license-file="$FS_LICENSE" --task-id="gas" --fs-no-reconall --use-aroma --output-spaces 'MNI152NLin6Asym:res-2' 'T1w' 'MNI152NLin6Sym' --participant-label 001`

@mgxd oh! that would explain things, although I thought fmriprep was taking care of the downsampling.

**Comment by oesteban** [comment]
@antotorov fMRIPrep can use the resolutions distributed by the template creators. It so happens that the original `MNI152NLin6Syn` does not have 2mm files (I believe).

**Comment by antotorov** [comment]
Ok thanks to all for your inputs! Those really are helpful.

I ran the same as above except I dropped the `:res-2` suffix. No errors where produced (according to the html report).

Looking into the outputs, however, I found something weird: the dimensions of
`sub-001_task-gas_space-MNI152NLin6Sym_desc-smoothAROMAnonaggr_bold.nii.gz`
and
`sub-001_task-gas_space-MNI152NLin6Sym_desc-brain_mask.nii.gz`
aren't the same! (91x109x91x32 for the first and 89x106x65x32 for the second).
By the way, running fsl_regfilt (as in https://fmriprep.readthedocs.io/en/stable/workflows.html#ica-aroma but using the series in MNI152NLin6Sym space as input, followed by 6mm FWHM smoothing) outputs an image with dimensions 89x106x65x32 as expected. So there is clearly something I am missing here... what do you think?

**Comment by effigies** [comment]
`sub-001_task-gas_space-MNI152NLin6Sym_desc-brain_mask.nii.gz` should be a 3d volume...

**Comment by jdkent** [comment]
This is a concept that confused me at first as well. The `smoothAROMAnonaggr` file is resliced (i.e., changing the sizes of the voxels) according to the voxel size in the MNI152NLin6Sym atlas as well as transformed into MNI152NLin6Sym space, whereas the latter file is only transformed and the voxel size remains the same as the original bold file you passed into fmriprep. Since fsl tools often do both reslicing and transforming, the difference between the two operations is blurred.

**Comment by antotorov** [comment]
@effigies my bad, I meant 89x106x65 for the mask and 89x106x65x32 for the `*space-MNI152NLin6Sym_desc-preproc_bold.nii.gz` series.

**Comment by oesteban** [comment]
@jdkent 
> This is a concept that confused me at first as well. The `smoothAROMAnonaggr` file is resliced (i.e., changing the sizes of the voxels) according to the voxel size in the MNI152NLin6Sym atlas as well as transformed into MNI152NLin6Sym space, whereas the latter file is only transformed and the voxel size remains the same as the original bold file you passed into fmriprep. Since fsl tools often do both reslicing and transforming, the difference between the two operations is blurred.

Do you know on top of your head whether we are moving the `smoothAROMAnonaggr` to other spaces via the T1w or if we are just resampling on the grid of all other spaces?

I'm very concerned about this, because option 1) does not seem recommendable - although correct, in principle. Then option 2) would be misleading - we should use the `res-X` flags should they be accepted in the common-derivatives (although I'd say we shouldn't even generate those).

If option 1), do you know how many interpolation steps are we running?

**Comment by jdkent** [comment]
@oesteban if I understand your question correctly, [currently we only provide](https://github.com/poldracklab/fmriprep/blob/b3a4d245bb0e625b0595c8bb53989b0b9b8fb4b2/fmriprep/workflows/bold/base.py#L893) `smoothAROMAnonaggr` in `MNI152NLin6Sym` space (resampled) and do not move the aroma results to any other space.

My much delayed pull request #1255's goal would run `fsl_regfilt` on all other spaces giving the user   `smoothAROMAnonaggr` in whatever output spaces they specified.

<details>
  <summary>Irrelevant Rambling</summary>
 which would then open us up to re-rerunning the confound pipeline on the "denoised" bold which would help circumvent the ambiguity around denoising strategies: I do really want to work on this, but thesis/work keeps getting in the way
</details>

I hope this answers your question!

**Comment by oesteban** [comment]
Judging by this:

https://github.com/poldracklab/fmriprep/blob/b3a4d245bb0e625b0595c8bb53989b0b9b8fb4b2/.circleci/ds005_outputs.txt#L70

we *do* generate denoised instances in seemingly different spaces. I have checked on the size though and they seem the same. Considering that you've been working on this for a long time now (#1255), I believe this is a bug introduced when I enabled the possibility of registering to several templates.

I've opened #1816 to follow up.

Regarding this issue, @antotorov, has your question been addressed?
